drivers/sensor/adxl375/adxl375.h

```diff
@@ -0,0 +1,348 @@
+#define ADXL375_FORMAT_JUSTIFY_MODE(x)    (((x)&0x1) << 2)
+
+/* ADXL375_FIFO_CTL */
+#define ADXL375_FIFO_CTL_FIFO_MODE_MSK GENMASK(7, 5)
+#define ADXL375_FIFO_CTL_TRIGGER_MSK   BIT(5)
+#define ADXL375_FIFO_CTL_SAMPLES_MSK   GENMASK(4, 0)
+
+#define ADXL375_FIFO_CTL_FIFO_MODE_MODE(x) (((x)&0x7) << 5)
+#define ADXL375_FIFO_CTL_TRIGGER_MODE(x)   (((x)&0x1) << 5)
+#define ADXL375_FIFO_CTL_SAMPLES_MODE(x)   ((x)&0x1F)
+
+/* ADXL375_FIFO_STATUS */
+#define ADXL375_FIFO_STATUS_FIFO_TRIG(x) (((x) >> 7) & 0x1)
+#define ADXL375_FIFO_STATUS_ENTRIES(x)   ((x)&0x7F)
+
+/* ADXL375 scale factors specified in page 3, table 1 of datasheet */
+#define ADXL375_MG2G_MULTIPLIER 0.049
+
+/* Set bits 3, 1 and 0 since register defaults to 0. Datasheet pages 20-24 */
+#define ADXL375_DATA_FORMAT_DEFAULT_BITS 0x0B
+
+enum adxl375_axis {
+	ADXL375_X_AXIS,
+	ADXL375_Y_AXIS,
+	ADXL375_Z_AXIS
+};
+
+enum adxl375_op_mode {
+	ADXL375_STANDBY = 0x04,
+	ADXL375_MEASUREMENT = 0x08,
+	ADXL375_AUTOSLEEP = 0x24
+};
+
+enum adxl375_bandwidth {
+	ADXL375_BW_0_05HZ,
+	ADXL375_BW_0_10HZ,
+	ADXL375_BW_0_20HZ,
+	ADXL375_BW_0_39HZ,
+	ADXL375_BW_0_78HZ,
+	ADXL375_BW_1_56HZ,
+	ADXL375_BW_3_13HZ,
+	ADXL375_BW_6_25HZ,
+	ADXL375_BW_12_5HZ,
+	ADXL375_BW_25HZ,
+	ADXL375_BW_50HZ,
+	ADXL375_BW_100HZ,
+	ADXL375_BW_200HZ,
+	ADXL375_BW_400HZ,
+	ADXL375_BW_800HZ,
+	ADXL375_BW_1600HZ
+};
+
+enum adxl375_odr {
+	ADXL375_ODR_0_10HZ = 0,
+	ADXL375_ODR_0_20HZ,
+	ADXL375_ODR_0_39HZ,
+	ADXL375_ODR_0_78HZ,
+	ADXL375_ODR_1_56HZ,
+	ADXL375_ODR_3_13HZ,
+	ADXL375_ODR_6_25HZ,
+	ADXL375_ODR_12_5HZ,
+	ADXL375_ODR_25HZ,
+	ADXL375_ODR_50HZ,
+	ADXL375_ODR_100HZ,
+	ADXL375_ODR_200HZ,
+	ADXL375_ODR_400HZ,
+	ADXL375_ODR_800HZ,
+	ADXL375_ODR_1600HZ,
+	ADXL375_ODR_3200HZ
+};
+
+enum adxl375_fifo_format {
+	ADXL375_XYZ_FIFO,
+	ADXL375_X_FIFO,
+	ADXL375_Y_FIFO,
+	ADXL375_XY_FIFO,
+	ADXL375_Z_FIFO,
+	ADXL375_XZ_FIFO,
+	ADXL375_YZ_FIFO,
+	ADXL375_XYZ_PEAK_FIFO,
+};
+
+enum adxl375_fifo_mode {
+	ADXL375_FIFO_BYPASS,
+	ADXL375_FIFO_STREAM,
+	ADXL375_FIFO_TRIGGER
+};
+
+struct adxl375_fifo_config {
+	enum adxl375_fifo_mode fifo_mode;
+	enum adxl375_fifo_format fifo_format;
+	uint16_t fifo_samples;
+};
+
+struct adxl375_activity_threshold {
+	uint16_t thresh;
+	bool referenced;
+	bool enable;
+};
+
+struct adxl375_xyz_accel_data {
+	int16_t x;
+	int16_t y;
+	int16_t z;
+};
+
+struct adxl375_transfer_function {
+	int (*read_reg_multiple)(const struct device *dev, uint8_t reg_addr, uint8_t *value,
+				 uint16_t len);
+	int (*write_reg)(const struct device *dev, uint8_t reg_addr, uint8_t value);
+	int (*read_reg)(const struct device *dev, uint8_t reg_addr, uint8_t *value);
+	int (*write_reg_mask)(const struct device *dev, uint8_t reg_addr, uint32_t mask,
+			      uint8_t value);
+};
+
+struct adxl375_data {
+	struct adxl375_xyz_accel_data sample;
+	const struct adxl375_transfer_function *hw_tf;
+	struct adxl375_fifo_config fifo_config;
+#ifdef CONFIG_ADXL375_TRIGGER
+	struct gpio_callback gpio_cb;
```

Comment:
trigger supports seems to be declared, but not implemented.

Reply:
In response to this comment and the one in the KConfig, no support for it yet. I wanted to get basic functionality for a driver in first and see about supporting it another time if I found time or if someone else was interested in doing so. I can remove this unused code.